### PR TITLE
Fix hesa ethnicity for 2023/2024 application forms

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

In this commit 05c8752f1f5beee512278f735029036b77eb9479 we have fixed a bug where the `Another White background` option to `ethnic_background` saved the wrong `hesa_ethnicity` code. It saved 160 rather than 179.

We now need to fix this issue for the old application_forms. There are about 5k application_forms in production, only for 2023 and 2024 recruitment cycles. This commit adds a data migration to fix this.

## Changes proposed in this pull request

Data migration

## Guidance to review

Does the query look right? Postgresql Jsonb documentation https://www.postgresql.org/docs/9.5/functions-json.html

`jsonb_set` should only change the specified key, `hesa_ethnicity`.

```
jsonb_set(target jsonb, path text[], new_value jsonb [, create_missing boolean])

Returns target with the section designated by path replaced by new_value, or with new_value
added if create_missing is true (default is true) and the item designated by path does not
exist.
As with the path oriented operators, negative integers that appear in path count from the
end of JSON arrays.
```

Blazer query https://www.apply-for-teacher-training.service.gov.uk/support/blazer/queries/1067-wrong-hesa_ethnicity-for-another-white-background

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Bug: DataMigration to fix hesa_ethnicity for "Another White background"](https://trello.com/c/Y8n0xheU/280-bug-datamigration-to-fix-hesaethnicity-for-another-white-background)

File: app/services/data_migrations/correct_hesa_ethnicity.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module DataMigrations
+  class CorrectHesaEthnicity
+    TIMESTAMP = 20241024110710
+    MANUAL_RUN = false
+
+    def change
+      application_forms.update_all(
```
Comment (dcyoung-dev) [inline app/services/data_migrations/correct_hesa_ethnicity.rb:7]: Is it worth doing this in batches? Not sure how many records we're updating here so might not be needed

File: spec/services/data_migrations/correct_hesa_ethnicity_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        equality_and_diversity: {
+          ethnic_group: 'White',
+          ethnic_background: 'British, English, Northern Irish, Scottish, or Welsh',
+          hesa_ethnicity: '160',
+        },
+      )
+
+      expect {
+        described_class.new.change
+      }.to change { form_2023.reload.equality_and_diversity['hesa_ethnicity'] }.from('160').to('179')
```
Comment (dcyoung-dev) [inline spec/services/data_migrations/correct_hesa_ethnicity_spec.rb:48]: Have we checked that other keys are not updated i.e `ethnic_background` is not updated or removed 😬.

File: app/services/data_migrations/correct_hesa_ethnicity.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        "equality_and_diversity = jsonb_set(equality_and_diversity, '{hesa_ethnicity}', '\"179\"', false)",
```
Comment (dcyoung-dev) [inline app/services/data_migrations/correct_hesa_ethnicity.rb:8]: `jsonb_set` is new to me TIL 🎓

**Review by dcyoung-dev — COMMENTED** [review]
This is a good bit of work 🚀 good spot on the `jsonb_set`

File: app/services/data_migrations/correct_hesa_ethnicity.rb (hunk 1 above)
Comment (CatalinVoineag) [reply app/services/data_migrations/correct_hesa_ethnicity.rb:7]: Just 5k so I think it's not necessary

File: spec/services/data_migrations/correct_hesa_ethnicity_spec.rb (hunk 2 above)
Comment (CatalinVoineag) [reply spec/services/data_migrations/correct_hesa_ethnicity_spec.rb:48]: Yeah, though I didn't include it in the spec, I'll include it

**Review by elceebee — APPROVED** [review]
Nice! I have learned something new here :)

**Pull request merged**